shared/events.js

```diff
@@ -262,24 +262,31 @@ export class GroupContract extends HashableContract {
     ['founderUsername', 'string']
   ])
   static vuex = GroupContract.Vuex({
-    state: { votes: [], payments: [], members: [], invitees: [] },
+    state: { votes: [], payments: [], members: [], invitees: [], profiles: {} },
     mutations: {
-      Payment (state, data) { state.payments.push(data) },
-      Vote (state, data) { state.votes.push(data) },
-      RecordInvitation (state, data) { state.invitees.push(data.username) },
-      DeclineInvitation (state, data) {
+      Payment (state, {data}) { state.payments.push(data) },
+      Vote (state, {data}) { state.votes.push(data) },
+      RecordInvitation (state, {data}) { state.invitees.push(data.username) },
+      DeclineInvitation (state, {data}) {
         let index = state.invitees.findIndex(username => username === data.username)
         if (index > -1) { state.invitees.splice(index, 1) }
       },
-      AcceptInvitation (state, data) {
+      AcceptInvitation (state, {data}) {
         let index = state.invitees.findIndex(username => username === data.username)
         if (index > -1) {
           state.invitees.splice(index, 1)
           state.members.push(data.username)
         }
       },
-      AcknowledgeFounder (state, data) {
+      AcknowledgeFounder (state, {data}) {
         if (!state.members.find(username => username === data.username)) { state.members.push(data.username) }
+      },
+      // TODO: remove group profile when leave group is implemented
+      ProfileAdjustment (state, {data}) {
+        let profile = state.profiles[data.username]
+        if (!profile) { profile = state.profiles[data.username] = {} }
+        if (!data.value) { return delete profile[data.name] }
+        profile[data.name] = data.value
```

Comment:
OK, can you rename this then to "SetGroupProfile" and rename the action class accordingly? Would make this much clearer.